### DISCUSSION TITLE
Cardio Vascular Disease mapping for the PoC

### DISCUSSION THREAD

**Discussion by hcadavid** [MyDigiTwinNL/MyDigiTwin-architecture-documentation#2] [General]
Hello @KasiaSmietanka 

I'm now creating the specification of the pairing rules for the FHIR CVD-condition resource based on the source code and the google docs document. The first case (based on the start_age variable in your code), which applies for heartfailure_presence_adu_q_2_a, heartfailure_presence_adu_q_2, and heartfailure_presence_adu_q_1 is quite clear to me. However, I'm not sure I understand how you are making the date approximation for the other cases (determine_event_date_using_year_or_regular_date in your code), that is, for the relevant variables with no related 'start_age', namely heartfailure_followup_adu_q_1 or cvd_followup_adu_q_1.

For other condition-pairing rules we worked on with @baukearends, we are using as an approximation the average between the assessment date where these follow-up variables have a ‘yes’ for the first time and the date of the preceding assessment. Considering this, the specification for this pairing rule is, for the moment, the following:

https://github.com/MyDigiTwinNL/CDF2Medmij-Mapping-tool/blob/f0ed12bb3beff08aa962b5a1e54f397b0d0b4cd0/src/lifelines/CardioVascularDisease.ts.wip#L48-L76

Would this be consistent with the way you approximate the event date? If not, could you describe the way you are approximating the event dates for the second case?

On the other hand, I couldn't find the promas_chronicdisease_adu_q_01_b variable on the Lifelines wiki, but according to the catalog, it was collected only on 2B. When the CVD status is determined only by this variable, how are you estimating the event date?

Thanks for your input on this!
Héctor

**Comment by KasiaSmietanka** [comment]
1. We are still finalising the outcomes definitions, but most of the mentioned variables will be needed. I will let you know when I have a final version of outcome definitions.
2. Event date approximation: as Bauke suggested, we can calculate the mean date between the last questionnaire date and the current one. The specific dates when questionnaires were filled are available in relevant files. For instance, the "heartfailure_followup_adu_q_1" was recorded in 1b, 1c (e.g., 1c_q_1_results.csv), 2a, 3a, 3b. To determine a proxy date for an event recorded in 1b, I propose calculating the mean between the assessment date and the 1b date. For events recorded in 1c, the proxy date would be the mean between the 1b event date and the 1c event date.
3. Re "promas_chronicdisease_adu_q_01_b" - I am not sure if we need this variable. I will let you know when we finalise CVD definition. However, if it is necessary, we could employ a similar approach as described in section 2 to ascertain a proxy event date. Assuming the questionnaires are labeled consecutively, we can utilize the date of the 2a assessment.

Please let me know if you have any additional questions.

**Reply by KasiaSmietanka** [reply]
Please find here the updated outcome definitions: https://docs.google.com/document/d/1GwuqRzA2GqbLXNWSnNedLhn7qoLA4znIDqL-os3lgVw/edit?usp=sharing

**Reply by hcadavid** [reply]
Hello @KasiaSmietanka and @baukearends 

Thanks for your answers and for sharing your discussion. The following is what I get from it (please let me know if I misunderstood something):

- The CVD condition would be considered active for a participant if they had an HF, MI, or Stroke conditions also reported as 'active'. Consequently the CVD on-set date would be the earliest on-set date of the three.

- The rules for 'mapping' Lifelines data with HF and MI conditions are quite clear, and they are consistent with the ones we already defined for [Stroke](https://github.com/MyDigiTwinNL/CDF2Medmij-Mapping-tool/blob/main/src/lifelines/Stroke.ts) one (e.g., using assessment dates for approximating incident stroke date)


However, if I read your discussion correctly, there is still no agreement on the inclusion of the `cvd_followup_adu_q_1` variable for inferring the CVD condition, right? 

My question is, wouldn't the information this question provides (_stroke, heart failure/heart muscle disorder, or clogged arteries in the legs "since the last time the questionnaire was filled"_) be already provided by the rules above mentioned defined for incident HI, MI, and Stroke? Or does it provides additional one we can use for inferring CVD condition? (e.g., intermittent claudication?) As @baukearends mentioned, we already discussed this but for the [vascular disease condition](https://github.com/MyDigiTwinNL/CDF2Medmij-Mapping-tool/issues/4#issuecomment-1600429111).



                                        [1A][1B][1C][2A][3A][3B]
  
    Stroke                                    
     stroke_startage_adu_q_1            [X ][  ][  ][  ][  ][  ]
     stroke_presence_adu_q_1            [X ][  ][  ][  ][  ][  ]
     stroke_followup_adu_q_1            [  ][X ][X ][X ][X ][X ]

    MI

     heartattack_startage_adu_q_1:      [X ][  ][  ][  ][  ][  ]    
     heartattack_presence_adu_q_1:      [X ][  ][  ][  ][  ][  ]    
     heartattack_followup_adu_q_1:      [  ][X ][X ][X ][X ][X ]    

    HF

     heartfailure_presence_adu_q_1      [X ][  ][  ][  ][  ][  ]    
     heartfailure_startage_adu_q_1      [X ][  ][  ][  ][  ][  ]    
     heartfailure_followup_adu_q_1      [  ][X ][X ][X ][X ][X ]    
	 
    CVD

     cvd_followup_adu_q_1:              [  ][  ][  ][X ][X ][X ] 
			> heart infarction
			> stroke (brain infarction, brain haemorrhage) (O)
			> heart failure/heart muscle disorder		   (O)
			> clogged arteries in the legs (intermittent claudication)

**Reply by KasiaSmietanka** [reply]
Re "The CVD condition would be considered active for a participant if they had an HF, MI, or Stroke conditions also reported as 'active'. Consequently the CVD on-set date would be the earliest on-set date of the three." -> yes, same for prevalent (before a baseline date) and incident (after a baseline date) disease

Please skip using `cvd_followup_adu_q_1`
